internal/controllers/gateway/route_predicates.go

```diff
@@ -87,13 +86,9 @@ func IsRouteAttachedToReconciledGateway[routeT gatewayapi.RouteT](
 			if isGatewayClassControlled(&gatewayClass) {
 				return true
 			}
-		default:
-			log.Error(
-				fmt.Errorf("unsupported parentRef kind %s and group %s", kind, group),
-				"Got an unexpected kind and group when checking route's parentRefs",
-			)
-			return false
 		}
+		// REVIEW: should we directly return false here if parentRef points to a non-Gateway object (like `Service`)?
+		// This means we do not reconcile the route when it is attaching to some non-Gateway parent, like using it for service mesh.
```

Comment:
I don't believe we should return here as there might be other parentRefs that are meant to be reconciled by KIC.

I'm not sure what would be the use case of attaching to a KIC and non KIC parent but that's a possibility.

Reply:
done.